src/test/kotlin/dev/stalla/model/factory/ItunesCategoryFactoryTest.kt

```diff
@@ -5,177 +5,115 @@ import assertk.assertThat
+            .filter { member -> member.visibility == KVisibility.PUBLIC }
+            .mapNotNull { member -> member.getter.call(this) }
+            .filterIsInstance<ItunesCategory>()
+            .toTypedArray()
     )
 
-    private val allCategoryNames = simpleCategoryNames + nestedCategoryNames
+    private val categoryPropertyMap: Map<String, ItunesCategory> by lazy {
+        val values: List<ItunesCategory> = ItunesCategory.Factory::class.declaredMemberProperties
+            .filter { member -> member.visibility == KVisibility.PUBLIC }
+            .mapNotNull { member -> member.getter.call(this) }
+            .filterIsInstance<ItunesCategory>()
 
-    @Test
-    fun `should retrieve all defined iTunes categories from the interface factory method`() {
-        for (simpleCategory in simpleCategoryNames) {
-            assertThat(ItunesCategory.of(simpleCategory)).isNotNull().all {
-                prop(ItunesCategory::type).isEqualTo(simpleCategory)
-                isInstanceOf(ItunesCategory.Simple::class)
-            }
+        values.associateBy({ it.type }, { it })
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(SimpleCategoryNameProvider::class)
+    fun `should retrieve all simple iTunes categories from the interface factory method`(simpleCategoryName: String) {
+        assertThat(ItunesCategory.of(simpleCategoryName)).isNotNull().all {
+            prop(ItunesCategory::type).isEqualTo(simpleCategoryName)
+            isInstanceOf(ItunesCategory.Simple::class)
         }
-        for (nestedCategory in nestedCategoryNames) {
-            assertThat(ItunesCategory.of(nestedCategory)).isNotNull().all {
-                prop(ItunesCategory::type).isEqualTo(nestedCategory)
-                isInstanceOf(ItunesCategory.Nested::class)
-            }
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(NestedCategoryNameProvider::class)
+    fun `should retrieve all nested iTunes categories from the interface factory method`(nestedCategoryName: String) {
+        assertThat(ItunesCategory.of(nestedCategoryName)).isNotNull().all {
+            prop(ItunesCategory::type).isEqualTo(nestedCategoryName)
+            isInstanceOf(ItunesCategory.Nested::class)
         }
     }
 
-    @Test
-    fun `should have a property in the interface factory companion for each iTunes category`() {
-        ItunesCategory.Factory::class.declaredMemberProperties.forEach { member ->
-            if (member.visibility == KVisibility.PUBLIC) {
-                val value = member.getter.call(this)
-                if (value is ItunesCategory) {
-                    assertThat(allCategoryNames).contains(value.type)
-                }
-            }
+    @ParameterizedTest
+    @ArgumentsSource(AllCategoryNameProvider::class)
+    fun `should retrieve all iTunes categories from the interface factory method`(categoryName: String) {
+        assertThat(ItunesCategory.of(categoryName)).isNotNull().all {
```

Comment:
Nit: you don't really need `all` here and in the following tests, it's only one assertion :)

Reply:
Just came back to say: in the next few files, too 😅

Reply:
True. Just a remnant from the copy pasting. Will simplify.